### PR TITLE
feat(commerce): clean up now replaced controllers

### PR DESCRIPTION
**Pull request by Spuffynism**
See `commerce.index.ts` to understand the scope of the changes. We remove top-level pagination, sort and facet generator controllers, in favor of those accessible through sub-controllers.

This PR is part of a series of cascading PRs: https://github.com/coveo/ui-kit/pull/3842

[CAPI-719]

[CAPI-719]: https://coveord.atlassian.net/browse/CAPI-719?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pvanool — APPROVED** [review]

**Review by olamothe — APPROVED** [review]

**Comment by olamothe** [comment]
Code delete is the best kind of PR to review

File: packages/headless/src/commerce.index.ts [hunk 1]
```diff
@@ -60,8 +60,6 @@ export type {
   PaginationProps,
   PaginationOptions,
 } from './controllers/commerce/core/pagination/headless-core-commerce-pagination';
-export {buildProductListingPagination} from './controllers/commerce/product-listing/pagination/headless-product-listing-pagination';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/commerce.index.ts:63]: These removed exports are obviously breaking, CC @spiracha-coveo

**Review by fbeaudoincoveo — APPROVED** [review]

**Review by louis-bompart — APPROVED** [review]

**Comment by Spuffynism** [comment]
@olamothe @louis-bompart @fbeaudoincoveo Some of the changes affect Atomic, and I'm not sure how to resolve them.

The atomic commerce pager for example uses the top-level pagination builders which are removed in this PR:
https://github.com/coveo/ui-kit/blob/d1ee1c3a9c19bd2d003c5f7bcd43a2481847bc96/packages/atomic/src/components/commerce/atomic-commerce-pager/atomic-commerce-pager.tsx#L92-L94

I'm thinking we should expose the listing/search controller on the commerce bindings directly. Would that be right?

**Comment by fbeaudoincoveo** [comment]
> @olamothe @louis-bompart @fbeaudoincoveo Some of the changes affect Atomic, and I'm not sure how to resolve them.
> 
> The atomic commerce pager for example uses the top-level pagination builders which are removed in this PR:
> 
> https://github.com/coveo/ui-kit/blob/d1ee1c3a9c19bd2d003c5f7bcd43a2481847bc96/packages/atomic/src/components/commerce/atomic-commerce-pager/atomic-commerce-pager.tsx#L92-L94
> 
> I'm thinking we should expose the listing/search controller on the commerce bindings directly. Would that be right?

Yes that could work and IMO it would make sense since you'd build the pagination, sort, and facets from those "parent" controllers...

E.g., you'd have:

```
public initialize() {
  if (this.bindings.interfaceElement.type === 'product-listing') {
    this.pager = this.bindings.productListing.pagination();
  } else if (this.bindings.interfaceElement.type === 'search') {
    this.pager = this.bindings.search.pagination();
  }
}
```

@olamothe what's your take on this?

**Comment by olamothe** [comment]
> > @olamothe @louis-bompart @fbeaudoincoveo Some of the changes affect Atomic, and I'm not sure how to resolve them.
> > The atomic commerce pager for example uses the top-level pagination builders which are removed in this PR:
> > https://github.com/coveo/ui-kit/blob/d1ee1c3a9c19bd2d003c5f7bcd43a2481847bc96/packages/atomic/src/components/commerce/atomic-commerce-pager/atomic-commerce-pager.tsx#L92-L94
> > 
> > I'm thinking we should expose the listing/search controller on the commerce bindings directly. Would that be right?
> 
> Yes that could work and IMO it would make sense since you'd build the pagination, sort, and facets from those "parent" controllers...
> 
> E.g., you'd have:
> 
> ```ts
> public initialize() {
>   if (this.bindings.interfaceElement.type === 'product-listing') {
>     this.pager = this.bindings.productListing.pagination();
>   } else if (this.bindings.interfaceElement.type === 'search') {
>     this.pager = this.bindings.search.pagination();
>   }
> }
> ```
> 
> @olamothe what's your take on this?


Let's do like this: https://github.com/coveo/ui-kit/pull/3895/files#diff-2fa6435e1e9a232914e7a52b08fc3d645170336854aba1f193646e220f23cdafR53-R60

**Comment by Spuffynism** [comment]
@olamothe @fbeaudoincoveo applied that suggestion in https://github.com/coveo/ui-kit/pull/3857/commits/d9932b7e03aeea974b76f4ae9bfe0b48a638f1c1

**Pull request merged**